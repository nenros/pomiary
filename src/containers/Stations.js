import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchStations } from '../actions/stations'

export class Stations extends Component {

    componentDidMount() {
        this.props.fetchStations()
    }

    render() {
        const { children } = this.props
        const { records, error } = this.props.stations
        if (error) {
            return <section className="message is-danger">
                <div className="message-header">
                    <p>Error</p>
                </div>
                <div className="message-body">
                    {error}
                </div>
            </section>
        }
        return React.cloneElement(children, { stations: records })
    }
}

export default connect(
    state => ({ stations: state.stations }),
    (dispatch) => ({ fetchStations: () => dispatch(fetchStations()) })
)(Stations)
