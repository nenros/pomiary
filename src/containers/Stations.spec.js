import React from 'react';
import { shallow } from 'enzyme';
import { Stations } from './Stations';

const fetchStations = jest.fn()
const records = []
const errorMessage = 'cos nie tak'
const defaultProps = {
    fetchStations,
    stations: {
        error: null,
        records
    },
    children: <div />
}
const setup = (props = defaultProps) => {
    return shallow(<Stations {...props} />)
}

describe('<Stations/> container', () => {
    afterEach(() => { fetchStations.mockClear() })

    it('should render', () => {
        const enzymeWrapper = setup()
        expect(enzymeWrapper).toMatchSnapshot()
    })

    it('should render children if there isn\'t error', () => {
        const enzymeWrapper = setup()
        expect(enzymeWrapper).toContainReact(<div stations={records} />)
    })

    it('should fetch stations at mount', () => {
        const enzymeWrapper = setup()
        expect(fetchStations.mock.calls).toHaveLength(1)
    })

    it('should render error', () => {
        const props = { ...defaultProps, stations: { ...defaultProps.stations, error: errorMessage } }
        const enzymeWrapper = setup(props)
        expect(enzymeWrapper.find('.message-body')).toHaveText(errorMessage)
        expect(enzymeWrapper).toMatchSnapshot()
    })

});
