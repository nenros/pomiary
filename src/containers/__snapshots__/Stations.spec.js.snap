// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stations/> container should render 1`] = `
<div
  stations={Array []}
/>
`;

exports[`<Stations/> container should render error 1`] = `
<section
  className="message is-danger"
>
  <div
    className="message-header"
  >
    <p>
      Error
    </p>
  </div>
  <div
    className="message-body"
  >
    cos nie tak
  </div>
</section>
`;
